### ISSUE TITLE
Python bindings

### ISSUE THREAD

**Issue by lavender04** [mfontanini/libtins#457]
Hi,

I'm currently writing a sniffer tool for my master thesis and stumbled over this awesome project. Due to the performance issues of pyshark and scapy on heavy load I would like to use this lib from my python code. I've read a bit into python bindings and asked myself whether there is someone in the community who has some hints what's the easiest way to build the bindings and call the lib from python code. Is it recommended to use one of the tools described [here](https://realpython.com/python-bindings-overview/#conclusion)?
Has someone experiences using [this](https://github.com/stephane-martin/cycapture)?

Any help is much appreciated.

Thanks a lot in advance for this awesome project.

**Comment by mfontanini** [comment]
I'd suggest either:
* Using a faster python library. According to the [benchmarks](https://libtins.github.io/benchmark/) I ran, [dpkt](https://dpkt.readthedocs.io/en/latest/) should be a lot faster than scapy. It definitely won't be as fast as libtins but it may give you enough of a performance boosts over the sluggish scapy.
* Write the code in C++ using libtins and export thiner bindings to whatever algorithm you have. e.g. if you were trying to build a traceroute tool, write it in C++ using libtins and then export a function such as `run_traceroute` so it is importable via python. I've used boost.python for this in the past and it works well, but there's probably better alternatives out there. This will let you write a smaller amount of code to glue the C++/Python worlds together as well as giving you some performance boost. If you end up having to wrap the entire library it's going to take you a very long time.

Hope this helps!
